### ISSUE TITLE
Third party plugin works in Emulator but not on actual device

### ISSUE THREAD

**Issue by snelson112** [BuildFire/sdk#345]
Hi,

I'm working on a church app that uses a custom plugin to display comments from our WP site.  Within the custom plugin, I'm using the following script to transform plain text bible references to a link that displays a popup of the bible verses within the comments retrieved.

`var script = document.createElement('script');
script.id = 'bw-highlighter-config';
script.setAttribute('data-autoparse', 'false');
script.setAttribute('data-version', 'NIV');
document.head.appendChild(script);

(function(w, d, s, e, id) {
  w._bhparse = w._bhparse || [];
  function l() {
    if (d.getElementById(id)) return;
    var n = d.createElement(s), x = d.getElementsByTagName(s)[0];
    n.id = id; n.async = true; n.src = '//bibles.org/linker/js/client.js';
    x.parentNode.insertBefore(n, x);
  }
  (w.attachEvent) ? w.attachEvent('on' + e, l) : w.addEventListener(e, l, false);
})(window, document, 'script', 'load', 'bw-highlighter-src');`

This works perfectly fine within the browser emulator but, not on my actual device using BuildFire Previewer app for Android.  Also, debugging with Weinre has proven difficult since the content of the custom plugin iframe is not visible.

Can someone please help me resolve this issue.

Thank you.

**Comment by DanielHindi** [comment]
My guess would be that you are referencing files without identifying the protocol 
# example:
` n.src = '//bibles.org/linker/js/client.js'`

This implies the use of the current protocol. For example on web you'll find `http` or `https` however you are not on the web when its in the device. which means its not using `http` protocols.
I believe if you change it to
` n.src = 'https://bibles.org/linker/js/client.js'`

this will solve your problem.

Hope this helps

**Comment by snelson112** [comment]
Thanks Daniel, unfortunately, changing the source to include either http or https does not resolve the issue.  I can provide the appID via email if you like?

**Comment by DanielHindi** [comment]
Can you share any of your console logs? Are you sure all your resources are loading on the device

> Also, debugging with Weinre has proven difficult since the content of the custom plugin iframe is not visible.

In regards to debugging with Weinre, you should see multiple remote sessions. One for the app, one for the home plugin and another for your plugin. If you select the correct one your context should resolve

**Comment by snelson112** [comment]
Thanks Daniel, 

I was able to add the debugger to the iframe within the custom plugin and that showed me that the additional files included after the js for the highlighter was loaded did not reference the correct protocols as you mentioned before.  I circumvented this by loading the additional files within my code.

I had to add the following to my code within the custom plugin in order for Weinre to show my session:

`var script = document.createElement('script');script.id = 'BuildFireAppDebuggerScript';
script.setAttribute('src', 'http://debug.buildfire.com/target/target-script-min.js#snelsontestapp');
document.head.appendChild(script);`

Thank you for assisting me with this issue!
